### PR TITLE
Bug 1761280: don't log secret data

### PR DESCRIPTION
**Pull request by soltysh**
/assign @tnozicka @sttts

### CODE REVIEW

File: pkg/operator/resource/resourceapply/core.go [hunk 1]
```diff
@@ -234,23 +234,7 @@ func ApplySecret(client coreclientv1.SecretsGetter, recorder events.Recorder, re
...
-				safeRequired.Data[s] = []byte("NEW")
-			} else if !equality.Semantic.DeepEqual(existing.Data[s], safeRequired.Data[s]) {
-				safeRequired.Data[s] = []byte("MODIFIED")
-			} else {
-				safeRequired.Data[s] = []byte("OLD")
-			}
-		}
-
-		klog.Infof("Secret %q changes: %v", required.Namespace+"/"+required.Name, JSONPatch(safeExisting, safeRequired))
+		klog.Infof("Secret %q changes: %v", required.Namespace+"/"+required.Name, JSONPatchSecret(existing, required))
```
Comment (tnozicka) [inline pkg/operator/resource/resourceapply/core.go]: nit: don't concatenate string while doing printf at the same time (`'%s/%s'`)

Comment (tnozicka) [reply pkg/operator/resource/resourceapply/core.go]: (inherited)

File: pkg/operator/staticpod/controller/revision/revision_controller.go [hunk 2]
```diff
@@ -193,7 +193,7 @@ func (c RevisionController) isLatestRevisionCurrent(revision int32) (bool, strin
 		}
 		if !equality.Semantic.DeepEqual(existingData, requiredData) {
 			if klog.V(4) {
-				klog.Infof("secret %q changes for revision %d: %s", s.Name, revision, resourceapply.JSONPatch(existing, required))
+				klog.Infof("Secret %q changes for revision %d: %s", s.Name, revision, resourceapply.JSONPatchSecret(existing, required))
```
Comment (tnozicka) [inline pkg/operator/staticpod/controller/revision/revision_controller.go]: preexisting but it si missing namespace

File: pkg/operator/resource/resourceapply/core.go (hunk 1 above)
Comment (soltysh) [reply pkg/operator/resource/resourceapply/core.go]: Good point, lemme quickly fix it.

File: pkg/operator/staticpod/controller/revision/revision_controller.go (hunk 2 above)
Comment (soltysh) [reply pkg/operator/staticpod/controller/revision/revision_controller.go]: `RevisionResource` does not hold Namespace, thus we don't print it.

**Pull request merged**